src/irmin-pack/IO.ml

```diff
@@ -117,28 +117,50 @@ module Unix : S = struct
 
   type t = {
     file : string;
-    mutable raw : Raw.t;
+    mutable raw : Raw.t option;
```

Comment:
Perhaps we should add a comment to explain the semantics of this option? Something like "may be `None` if no read-write instances have been constructed".

(I thought about suggesting a type that only permits a `None` → `Some x` transition, but the comment probably suffices)